hedera-node/hapi-utils/src/main/java/com/hedera/node/app/hapi/utils/ethereum/EthTxData.java

```diff
@@ -205,6 +205,12 @@ public BigInteger getMaxGasAsBigInteger() {
      * @return the effective offered gas price
      */
     public long effectiveOfferedGasPriceInTinybars() {
+
+        // prevent division by zero
+        if (gasLimit == 0) {
```

Comment:
Is it possible to call this with `gasLimit = 0`? If so, what test surfaced that?

Reply:
`nonceNotUpdatedWhenInsufficientIntrinsicGasPrecheckFailed` catches this edge case. WIthout the above check it fails with `FAIL_INVALID`

Reply:
Would it make more sense then to avoid the division by zero on L#214? I think it's confusing to return an offered gas **price** of zero just because the gas **limit** is zero.

I don't know what the point of dividing by `gasLimit` was in the first place. Seems like we can just do,
```
        return BigInteger.valueOf(Long.MAX_VALUE)
                .min(getMaxGasAsBigInteger().divide(WEIBARS_TO_TINYBARS))
                .longValueExact();
```
Which will result in `HederaEvmTransaction.offeredGasCost()` [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/hevm/HederaEvmTransaction.java#L119) always overflowing for a non-zero `gasLimit`.